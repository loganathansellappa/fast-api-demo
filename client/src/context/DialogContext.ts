import { Dispatch } from 'react';
import { createContext, SetStateAction, useContext } from 'react';

export type DialogState = {
  isOpen: boolean;
  content: React.ReactNode | null;
  handler: () => void;
  title: string;
  noBtnText: string;
  yesBtnText: string;
};

export type DialogContextType = {
  dialog: DialogState;
  setDialog: Dispatch<SetStateAction<DialogState>>;
};

export const DialogContext = createContext<DialogContextType | null>(null);


export const useDialogContext = () => {
  /* eslint-disable  @typescript-eslint/no-explicit-any */
  const context: any = useContext(DialogContext)
  if (context === undefined) {
    throw new Error('useAxiosClient must be used within a AxiosClientProvider')
  }
  return context;
}