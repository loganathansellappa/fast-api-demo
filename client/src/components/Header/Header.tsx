import { useDialogContext } from '../../context/DialogContext.ts';
import { CreateMissionControlForm } from '../CreateMissionControlForm/CreateMissionControlForm.tsx';
import { useCallback } from 'react';

export const Header = () => {
  const { setDialog } = useDialogContext();

  const closeDialog = useCallback(() => {
    setDialog({
      isOpen: false,
      text: "",
      handler: null,
      noBtnText: "",
      yesBtnText:""
    });
  }, [setDialog]);

  const openDialog = useCallback(() => {
    setDialog({
      isOpen: true,
      title: 'Add - Mission',
      content: <CreateMissionControlForm  onCancel={closeDialog} />
    });
  }, [setDialog, closeDialog]);

    return (
      <div className="flex">
        <h1 className="bg-gray-100 text-black font-bold py-2 px-4 rounded"
        >Flight Mission Control</h1>
        <button
          className="bg-gray-200 hover:bg-gray-300 text-black font-bold py-2 px-4 rounded ml-auto"
          onClick={() => openDialog()}
        >
          Add Mission Control
        </button>
      </div>
    );
}