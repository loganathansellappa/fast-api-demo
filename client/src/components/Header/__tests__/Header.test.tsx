import { render } from "@testing-library/react";
import { Header } from '../Header.tsx';
import { createContext } from 'react';
import * as Contexts from '../../../context/DialogContext.ts';

let element: { container: HTMLElement };

beforeEach(() => {


  const context = createContext<Contexts.DialogContextType>({
    dialog: {
      isOpen: false,
      content: null,
      handler: () => jest.fn(),
      title: "",
      noBtnText: "",
      yesBtnText: "",
    },
    setDialog: () => {}
  })
  jest
    .spyOn(Contexts, "useDialogContext")
    .mockReturnValue(context);
  element = render(<Header />);
});

describe("Header", () => {

  test("renders Header component", () => {
    expect(element.container.firstChild).toMatchSnapshot();
  });

  test("displays title and button", () => {
    expect(element.container.getElementsByTagName('h1')[0].textContent).toContain("Flight Mission Control");
    expect(element.container.getElementsByTagName('button')[0].textContent).toContain("Add Mission Control");
  });
});
