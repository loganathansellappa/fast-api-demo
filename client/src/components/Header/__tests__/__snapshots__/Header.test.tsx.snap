// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
<div
  class="flex"
>
  <h1
    class="bg-gray-100 text-black font-bold py-2 px-4 rounded"
  >
    Flight Mission Control
  </h1>
  <button
    class="bg-gray-200 hover:bg-gray-300 text-black font-bold py-2 px-4 rounded ml-auto"
  >
    Add Mission Control
  </button>
</div>
`;
